Source/VMware.vSphereDSC/DSCResources/VMHostIScsiHba.ps1

```diff
@@ -0,0 +1,126 @@
+#>
+
+[DscResource()]
+class VMHostIScsiHba : VMHostIScsiHbaBaseDSC {
+    <#
+    .DESCRIPTION
+
+    Specifies the name of the iSCSI Host Bus Adapter.
+    #>
+    [DscProperty(Key)]
+    [string] $Name
+
+    hidden [string] $ConfigureIScsiHbaChapMessage = "Configuring CHAP settings of iSCSI Host Bus Adapter {0} from VMHost {1}."
+
+    hidden [string] $CouldNotConfigureIScsiHbaChapMessage = "Could not configure CHAP settings of iSCSI Host Bus Adapter {0} from VMHost {1}. For more information: {2}"
+
+    [void] Set() {
+        try {
+            Write-VerboseLog -Message $this.SetMethodStartMessage -Arguments @($this.DscResourceName)
+            $this.ConnectVIServer()
+            $this.RetrieveVMHost()
+
+            $iScsiHba = $this.GetIScsiHba($this.Name)
+
+            $this.ConfigureIScsiHbaChap($iScsiHba)
+        }
+        finally {
+            $this.DisconnectVIServer()
+            Write-VerboseLog -Message $this.SetMethodEndMessage -Arguments @($this.DscResourceName)
+        }
+    }
+
+    [bool] Test() {
+        try {
+            Write-VerboseLog -Message $this.TestMethodStartMessage -Arguments @($this.DscResourceName)
+            $this.ConnectVIServer()
+            $this.RetrieveVMHost()
+
+            $iScsiHba = $this.GetIScsiHba($this.Name)
+
+            $result = !$this.ShouldModifyCHAPSettings($iScsiHba.AuthenticationProperties, $null, $null)
+
+            $this.WriteDscResourceState($result)
+
+            return $result
+        }
+        finally {
+            $this.DisconnectVIServer()
+            Write-VerboseLog -Message $this.TestMethodEndMessage -Arguments @($this.DscResourceName)
+        }
+    }
+
+    [VMHostIScsiHba] Get() {
+        try {
+            Write-VerboseLog -Message $this.GetMethodStartMessage -Arguments @($this.DscResourceName)
+            $result = [VMHostIScsiHba]::new()
+
+            $this.ConnectVIServer()
+            $this.RetrieveVMHost()
+
+            $iScsiHba = $this.GetIScsiHba($this.Name)
+
+            $this.PopulateResult($result, $iScsiHba)
+
+            return $result
+        }
+        finally {
+            $this.DisconnectVIServer()
+            Write-VerboseLog -Message $this.GetMethodEndMessage -Arguments @($this.DscResourceName)
+        }
+    }
+
+    <#
+    .DESCRIPTION
+
+    Configures the CHAP properties of the specified iSCSI Host Bus Adapter.
+    #>
+    [void] ConfigureIScsiHbaChap($iScsiHba) {
+        $setVMHostHbaParams = @{
+            IScsiHba = $iScsiHba
+            Confirm = $false
+            ErrorAction = 'Stop'
+            Verbose = $false
+        }
+
+        $this.PopulateCmdletParametersWithCHAPSettings($setVMHostHbaParams, $null, $null)
```

Comment:
An overload may be added without last two parameters

Reply:
Method overload was added to ShouldModifyCHAPSettings() and PopulateCmdletParametersWithCHAPSettings().